libxlsxwriter: 0.7.3 (new formula)

jmcnamara opened the pull request:
Add a new formula for libxlsxwriter.

libxlsxwriter is a C library for writing files in the Excel 2007+
XLSX file format.

It can be used to write text, numbers, formulas and hyperlinks to
multiple worksheets and it supports features such as formatting,
images, charts, page setup and many others.

See: https://libxlsxwriter.github.io/index.html

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,35 @@
+class Libxlsxwriter < Formula
+  desc "Libxlsxwriter: A C library for creating Excel XLSX files."
```
Shorten it to _“C library for creating Excel .xlsx files”_

jmcnamara replied:
Is "A C library for creating Excel .xlsx files" okay? That is the phrase I use in the docs.

fxcoudert replied:
No article (“A”) at the start please

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,35 @@
+  url "https://github.com/jmcnamara/libxlsxwriter/archive/RELEASE_0.7.3.tar.gz"
+  sha256 "84dbbd164b440695119ded3a8e2b6cac07d0d284343ab977c88dc0418d8fcde5"
+
+  depends_on "lzlib"
```
[Web page](https://github.com/jmcnamara/libxlsxwriter) says zlib is the only dependency. Why `lzlib` here?

jmcnamara replied:
Is that not the same thing or am I missing something?

fxcoudert replied:
_zlib_ vs _**l**zlib_

ilovezfs replied:
And zlib is included with macOS so the dependency should just be dropped.

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,35 @@
+
+  def install
+    lib.mkdir
+    include.mkdir
```
These directories should really be created by the makefile itself

jmcnamara replied:
Yes. Agreed. I will fix that.

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,35 @@
+    EOS
+
+    system ENV.cc, "test.c", "-L#{lib}", "-I#{include}", "-lxlsxwriter", "-o", "test"
+    system "./test"
```
Test is great! Please also add a final check that the file created actually exists, after running `test`

jmcnamara replied:
Will do.

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,35 @@
+    lib.mkdir
+    include.mkdir
+    system "make", "V=1"
+    system "make", "install", "INSTALL_DIR=#{prefix}", "V=1"
```
How about using `cmake` (which is present) instead of this two-step procedure?

jmcnamara replied:
`make` is the secondary build system for people who prefer it. The make build should really be 1 step as well but there is an error that I didn't notice until I was preparing this formula. Once it is fixed the following will be sufficient:

    system "make", "install", "INSTALL_DIR=#{prefix}", "V=1"

That will need a respin of libxlsxwriter to 0.7.4 though.

jmcnamara commented:
Thanks for the review and for your work maintaining brew. I'll address/reply to your comments.

jmcnamara commented:
I've made the request changes and pushed them. Tests pass locally and in CI.

Note, I had to do an upstream release of libxlsxwriter (0.7.4) so the title of this PR is no longer correct, although the commit message is correct. Let me know if you want me to close this and resubmit another PR.

fxcoudert commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,31 @@
+class Libxlsxwriter < Formula
+  desc "C library for creating Excel XLSX files."
```
No final `.`

ilovezfs commented on Formula/libxlsxwriter.rb:
```diff
@@ -0,0 +1,31 @@
+
+    system ENV.cc, "test.c", "-L#{lib}", "-I#{include}", "-lxlsxwriter", "-o", "test"
+    system "./test"
+    assert File.exist? "./myexcel.xlsx"
```
should use assert_predicate and a message

jmcnamara commented:
Okay, I've made the following requested changes:

- [x] Removed colon from commit subject line.
- [x] Removed trailing full stop from Formula description.
- [x] Used `assert_predicate()` instead of `assert()` and tested a failing test case. The fail output looks better with `assert_predicate()`, thanks for the suggestion.

Hopefully everything is okay now. But if not let me know. :-)

ilovezfs commented:
@jmcnamara Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
